### PR TITLE
Add experimental Enum.entries and update examples

### PR DESCRIPTION
**Pull request by sarahhaggarty**
Explain the new `entries` property for Enum classes. This page needs restructuring but I plan to do this in a separate PR to save time on review.

### CODE REVIEW

File: docs/topics/enum-classes.md [hunk 1]
```diff
@@ -87,11 +87,23 @@ EnumClass.valueOf(value: String): EnumClass
...
 ```
 
+Below is an example of these methods in action:
+
+```kotlin
+enum class RGB { RED, GREEN, BLUE }
+
+fun main() {
+    for (color in RGB.values()) println(color.toString()) // prints RED, GREEN, BLUE
+    println("The first color is: " + RGB.valueOf("RED")) // prints "The first color is: RED"
```
Comment (zarechenskiy) [inline docs/topics/enum-classes.md:97]: Minor: `println("The first color is: ${RGB.valueOf("RED")}")`

File: docs/topics/enum-classes.md [hunk 2]
```diff
@@ -103,9 +115,41 @@ inline fun <reified T : Enum<T>> printAllValues() {
...
+enum class RGB { RED, GREEN, BLUE }
+
+@OptIn(ExperimentalStdlibApi::class)
+fun main() {
+    for (color in RGB.entries) println(color.toString())
+    // prints RED, GREEN, BLUE
+}
+```
+
+> The `entries` property is [Experimental](components-stability.md#stability-levels-explained). To use it you need
```
Comment (zarechenskiy) [inline docs/topics/enum-classes.md:133]: Hmm, there is a problem that this compatibility guide is a bit incorrect and confuses technologies such as Kotlin/JVM and certain language features which should have their own definitions of compatibility levels. We are thinking about how to redesign it to include correct information about language features. I'd propose leaving the text as is, but remove the link to this guide, there should be enough explanation in `ExperimentalStdlibApi` annotation

**Review by zarechenskiy — CHANGES_REQUESTED** [review]
Looks good, only minor notes, thank you!

File: docs/topics/enum-classes.md [hunk 3]
```diff
@@ -103,9 +115,41 @@ inline fun <reified T : Enum<T>> printAllValues() {
 printAllValues<RGB>() // prints RED, GREEN, BLUE
 ```
 
-Every enum constant has properties for obtaining its name and position (starting with 0) in the enum class declaration:
+If you don't want your enum constants returned in an array, you can use the `entries` property instead. The
```
[the remark this answers is not in the document]
Comment (sarahhaggarty) [reply docs/topics/enum-classes.md]: Thanks @zarechenskiy! Can you check if the updated text is OK now?

**Review by zarechenskiy — APPROVED** [review]
👍

**Pull request closed without merging**